devito/archinfo.py

```diff
@@ -1,13 +1,15 @@
 """Collection of utilities to detect properties of the underlying architecture."""
 
-from subprocess import PIPE, Popen
+from subprocess import PIPE, Popen, DEVNULL
 
 import numpy as np
 import cpuinfo
 import psutil
 
+from cached_property import cached_property
```

Comment:
this one needs to be together with the third-party imports, such as numpy, cpuinfo, psutil, not with the devito imports